packages/rocketchat-sharedsecret/sharedsecret.js

```diff
@@ -0,0 +1,87 @@
+const SharedSecret = [];
+
+if (Meteor.isServer) {
+	const EncryptMessage = ((message) => {
+		const currentUser = Meteor.user()._id;
+		if (message) {
+			const currentRoomId = message.rid;
+			if (SharedSecret && SharedSecret[currentUser] && SharedSecret[currentUser][currentRoomId]) {
+				const currentSecret = SharedSecret[currentUser][currentRoomId];
+				const encrypted = CryptoJS.AES.encrypt(message.msg, currentSecret);
+				message.msg = encrypted.toString();
+				if (message.urls) {
+					const messageUrls = message.urls;
+					messageUrls.forEach((i) => {
+						const urls = messageUrls[i];
```

Comment:
The `i` is the `urls` already. Just rename the `i` to `urls` and remove the `const urls = messageUrls[i];`